### ISSUE TITLE
INJECTION_WEBHOOK_CONFIG_NAME is incorrect when in non default namespace

### ISSUE THREAD

**Issue by timmyers** [istio/istio#22808]
**Bug description**
The `INJECTION_WEBHOOK_CONFIG_NAME` env var set for `istiod` deployment is incorrect when the `IstioOperator` CR is deployed in a namespace other than `istio-system`.

This results in the webhook never getting patched with the caBundle, sidecar injection failing, and no pods being created.

**Expected behavior**

Env var should match the actual mutating webhook configuration name, it should be patched successfully, and pods should be created.

**Steps to reproduce the bug**
- Install standalone operator 1.5.x
- Create CR such that control plane will be created in non-default namespace, i.e. `istio`.
- Mutating webhook configuration is created with name `istio-sidecar-injector-${namespace}`, i.e. `istio-sidecar-injector-istio`.
- `istiod` deployment is crewated with `INJECTION_WEBHOOK_CONFIG_NAME` set to `istio-sidecar-injector`
- `istiod` fails to patch mutating webhook configuration

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
❯ istioctl version --remote
client version: 1.5.1
control plane version: 1.5.1
data plane version: 1.5.1 (3 proxies)

❯ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-14T04:24:34Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.10-eks-bac369", GitCommit:"bac3690554985327ae4d13e42169e8b1c2f37226", GitTreeState:"clean", BuildDate:"2020-02-26T01:12:54Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
By applying operator 1.5.1 manifest manually

**Environment where bug was observed (cloud vendor, OS, etc)**
EKS 1.15

**Comment by timmyers** [comment]
This can technically be considered fixed by https://github.com/istio/istio/pull/22828, however #22812 is still in flux and related as well.

**Comment by howardjohn** [comment]
Fixed by #22828. Thanks!
